src/context.cpp

```diff
@@ -138,15 +138,27 @@ memory::vector<event> context::match(const memory::unordered_set<rule *> &rules_
             ruleset_->dynamic_processors, deadline);
     };
 
+    // Evaluate user priority collections first
+    for (auto &[type, collection] : ruleset_->user_priority_collections) {
+        DDWAF_DEBUG("Evaluating user priority collection %s", type.data());
+        eval_collection(type, collection);
+    }
+
     // Evaluate priority collections first
-    for (auto &[type, collection] : ruleset_->priority_collections) {
+    for (auto &[type, collection] : ruleset_->base_priority_collections) {
         DDWAF_DEBUG("Evaluating priority collection %s", type.data());
```

Comment:
I'm not sure about these logging statements. It's unclear if `type.data()` is always nul terminated or even if `data` is not null. In particular, `get_tag` can return an empty `string_view`.

Reply:
I think we've discussed this before as well...

Reply:
To clarify, the `type` tag always needs to be present as it's enforced by the parser, so the only real concern is the nul-termination.

Reply:
Using `%.*s` format now.